base/stats.go

```diff
@@ -65,39 +66,52 @@ type SgwStats struct {
 	dbStatsMapMutex sync.Mutex
 }
 
-var SyncGatewayStats SgwStats
+var SyncGatewayStats *SgwStats
 
 // SkipPrometheusStatsRegistration used to avoid registering a stat with prometheus
 // Defaults to false - Only intended for test use
 // Note that due to NewSyncGatewayStats() being ran as part of init the initial stats will still be registered and so
 // global stats are unaffected
 var SkipPrometheusStatsRegistration bool
 
-func NewSyncGatewayStats() *SgwStats {
+func NewSyncGatewayStats() (*SgwStats, error) {
 	sgwStats := SgwStats{
 		GlobalStats: &GlobalStat{},
 		DbStats:     map[string]*DbStats{},
 	}
 
-	sgwStats.GlobalStats.initResourceUtilizationStats()
-	sgwStats.initReplicationStats()
+	err := sgwStats.GlobalStats.initResourceUtilizationStats()
+	if err != nil {
+		return nil, err
+	}
+	err = sgwStats.initReplicationStats()
+	if err != nil {
+		return nil, err
+	}
 
 	// This provides a stat for sgw_up where the value will be fixed to one. This is to allow backwards compatibility
 	// where the standalone exporter would export a value of 1 if it has contact with SGW.
-	NewIntStat("", "up", nil, nil, prometheus.GaugeValue, 1)
+	_, err = NewIntStat("", "up", nil, nil, prometheus.GaugeValue, 1)
+	if err != nil {
+		return nil, err
+	}
 
-	return &sgwStats
+	return &sgwStats, nil
 }
 
 func init() {
 	// Initialize Sync Gateway Stats
 
 	// All stats will be stored as part of this struct. Global variable accessible everywhere. To add stats see stats.go
-	SyncGatewayStats = *NewSyncGatewayStats()
+	sgwStats, err := NewSyncGatewayStats()
+	if err != nil {
+		log.Fatalf("Fatal error: %v", err)
```

Comment:
Add a little more context. Maybe there is more context lower but nice to have. 
`Fatal error attempting to instantiate sync gateway stats: %v`. Something like that?